### PR TITLE
IGNITE-25458 Fix deletion of renamed configuration from the storage

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-25458

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java [hunk 1]
```diff
@@ -940,6 +942,80 @@ public T2 next() {
...
+    /**
+     * Removes {@code null} values that correspond to non-deprecated legacy keys from the configuration tree.
+     *
+     * @param roots Super root.
+     * @param prefixMap Mutable prefix map with updates received from the storage.
+     * @see PublicName#legacyNames()
+     */
+    public static void ignoreLegacyKeys(SuperRoot roots, Map<String, ?> prefixMap) {
+        roots.traverseChildren(new KeysTrackingConfigurationVisitor<>() {
+            private Map<String, ?> map = prefixMap;
```
Comment (Phillippko) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java]: This field seems redundant, and has worse naming, let's remove it?

**Review by Phillippko — APPROVED** [review]
lgtm

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java (hunk 1 above)
Comment (ibessonov) [reply modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java]: It's not redundant, I use it for recursion. I'll give it a better name
